### DISCUSSION TITLE
ORAS v1.2.0 milestone discussion

### DISCUSSION THREAD

**Discussion by FeynmanZhou** [oras-project/oras#1119] [General]
ORAS is following a quarter-based release cadence in 2023. ORAS v1.1.0 was released on Sep 5. It's time to determine the v1.2.0 release plan and kick off the work for v1.2.0  in the community. After triaging all issues, I grouped the major enhancements into five aspects as follows. Some enhancements are suggested as critical and some are nice to have.

Based on the quarterly release cycle, the release date could be set to the middle of December.

This is an open discussion and I am also open to any suggestions. Feel free to share anything feature requests you think are missing below or should be prioritized.

### Standardize the CLI output for scripting experience and improve output experience

Provide a progress bar to improve the usability of content transportation. Standardize the CLI output for scripting experience.

- https://github.com/oras-project/oras/issues/893
- https://github.com/oras-project/oras/issues/940
- https://github.com/oras-project/oras/issues/638

### Improve verbose logs for all ORAS commands to enhance the troubleshooting experience

Make the verbose logs consistent, useful, concise, and human-readable. Users can easily locate the root cause by analyzing the verbose logs. Formulize the CLI output guideline to provide a unified reference for future command output.

   - https://github.com/oras-project/oras/issues/911
   - https://github.com/oras-project/oras/issues/1091
   - https://github.com/oras-project/oras/issues/916

This issue #1053 relies on https://github.com/oras-project/oras-go/issues/576

### Simplify image lifecycle metadata management experience through annotations

Users can easily update the image lifecycle metadata without learning the OCI Spec. Annotating experience is critical to users who want to attach or add image lifecycle metadata. The annotating functionality is useful for software supply chain scenarios such as adding "End-of-support" metadata to an OCI image.

- https://github.com/oras-project/oras/issues/571

### Optimize ORAS performance 

Decrease the auth request time to optimize ORAS performance.

- https://github.com/oras-project/oras/issues/1105

This issue relies on https://github.com/oras-project/oras-go/issues/594

### Support multi-arch image index management (Nice to have)

Most of the open-source projects support multi-arch images. Enable users to push/attach a multi-arch image or an image index by leveraging the [OCI image index](https://github.com/opencontainers/image-spec/blob/main/image-index.md).

- https://github.com/oras-project/oras/issues/1066
- https://github.com/oras-project/oras/issues/1053
- https://github.com/oras-project/oras/issues/813
   
### Complement OCI layout management (Nice to have)

ORAS has a few deletion capabilities for some ORAS commands that are not supported in the OCI image layout. Complement OCI layout management enables users to better manage the OCI image layout.

- https://github.com/oras-project/oras/issues/761
- https://github.com/oras-project/oras/issues/762

These two issues rely on https://github.com/oras-project/oras-go/issues/454

### Automate installation package manager (Winget, Snap) management (Nice to have)

- https://github.com/oras-project/oras/issues/1025

**Comment by FeynmanZhou** [comment]
We agreed to move https://github.com/oras-project/oras/issues/816 to milestone v1.2.0 based on the community meeting discussion on Sep 13, 2023.

**Comment by TerryHowe** [comment]
LGTM

**Comment by Amulyam24** [comment]
Hi! Is there an estimated time period for releasing 1.2.0?

**Reply by FeynmanZhou** [reply]
Hi @Amulyam24 ,

Based on the development process of the [v1.2.0 milestone](https://github.com/oras-project/oras/milestone/17), the ETA release date will be around mid of Jan. 
Do you have any additional requests on this release?

**Reply by Amulyam24** [reply]
Hi @FeynmanZhou. Thank you for the update!
I'm looking forward to `ppc64le` artefacts being released after the support has been added in #1118.

**Reply by FeynmanZhou** [reply]
Hi @Amulyam24 , ORAS v1.2.0-beta.1 released on Feb 8 and has the `ppc64le` release binary:  https://github.com/oras-project/oras/releases/download/v1.2.0-beta.1/oras_1.2.0-beta.1_linux_ppc64le.tar.gz

**Reply by Amulyam24** [reply]
Thank you @FeynmanZhou !

**Comment by Amulyam24** [comment]
Hi @FeynmanZhou, what is the expected timeline for the release of 1.2.0?

**Reply by FeynmanZhou** [reply]
@Amulyam24  The stable release of ORAS v1.2.0 is planned to be released at the beginning of Apr.

**Reply by q2go** [reply]
Any news regarding v1.2.0? It's middle of April already. Are there any unresolved problems in [v1.2.0-beta.1](https://github.com/oras-project/oras/releases/tag/v1.2.0-beta.1) which might even require another beta release?

**Reply by FeynmanZhou** [reply]
Hi @q2go , apologies for missing your question. Now the latest stable release is v1.2.2 https://github.com/oras-project/oras/releases/tag/v1.2.2
